src/shared_modules/dbsync/tests/interface/dbsync_test.cpp

```diff
@@ -383,7 +383,71 @@ TEST_F(DBSyncTest, GetDeletedRowsInvalidInput)
 {
     CallbackMock wrapper;
     callback_data_t callbackData { callback, &wrapper };
-    EXPECT_NE(0, dbsync_get_deleted_rows(nullptr, callbackData));
+    EXPECT_NE(0, dbsync_get_deleted_rows(nullptr, callbackData, nullptr));
+}
+
+TEST_F(DBSyncTest, GetDeletedRowsOnlyPKs)
+{
+    const auto sql{ "CREATE TABLE processes(`pid` BIGINT, `name` TEXT, `euser` TEXT, PRIMARY KEY (`pid`,`name`)) WITHOUT ROWID;"};
+    const auto table { R"({"table":"processes"})" };
```

Comment:
We could use our `Query` builder classes for this.  See the `syncRowInsertAndModifiedWithOldData` test.

Reply:
I will apply this suggestion in C++ interface because it will only create overhead un C interface due:
- Current approach -> std::string -> cJSON by cJSON_Parse
- Suggested approach -> std::string->nlohmann::json by parse()->std::string by dump -> cJSON by cJSON_Parse
What do you think?

Reply:
I would agree if we weren't talking about tests. But I think it's a good thing to go with the second approach because it somehow covers more of our code. If one of those things break, we will be able to tell right away.